lms/static/js/learner_dashboard/views/course_entitlement_view.js

```diff
@@ -0,0 +1,395 @@
+        'js/learner_dashboard/models/course_entitlement_model',
+        'js/learner_dashboard/models/course_card_model',
+        'text!../../../templates/learner_dashboard/course_entitlement.underscore',
+        'text!../../../templates/learner_dashboard/verification_popover.underscore'
+    ],
+         function(
+             Backbone,
+             $,
+             _,
+             gettext,
+             moment,
+             HtmlUtils,
+             EntitlementModel,
+             CourseCardModel,
+             pageTpl,
+             verificationPopoverTpl
+         ) {
+             return Backbone.View.extend({
+                 tpl: HtmlUtils.template(pageTpl),
+                 verificationTpl: HtmlUtils.template(verificationPopoverTpl),
+
+                 events: {
+                     'change .session-select': 'updateEnrollBtn',
+                     'click .enroll-btn': 'handleEnrollChange',
+                     'keydown .final-confirmation-btn': 'handleVerificationPopoverA11y',
+                     'click .popover-dismiss': function() { this.hideDialog($('.enroll-btn-initial')); }
+                 },
+
+                 initialize: function(options) {
+                     // Set up models and reload view on change
+                     this.courseCardModel = new CourseCardModel();
+                     this.entitlementModel = new EntitlementModel({
+                         availableSessions: this.formatDates(JSON.parse(options.availableSessions)),
+                         entitlementUUID: options.entitlementUUID,
+                         currentSessionId: options.currentSessionId,
+                         userId: options.userId
+                     });
+                     this.listenTo(this.entitlementModel, 'change', this.render);
+
+                     // Grab URLs that handle changing of enrollment and entering a newly enrolled session.
+                     this.enrollUrl = options.enrollUrl;
+                     this.courseHomeUrl = options.courseHomeUrl;
+
+                     // Grab elements from the parent card that work with this view and bind associated events
+                     this.$triggerOpenBtn = $(options.triggerOpenBtn); // Opens/closes session selection view
+                     this.$dateDisplayField = $(options.dateDisplayField); // Displays current session dates
+                     this.$enterCourseBtn = $(options.enterCourseBtn); // Button link to course home page
+                     this.$courseCardMessages = $(options.courseCardMessages); // Additional session messages
+                     this.$courseTitleLink = $(options.courseTitleLink); // Title link to course home page
+                     this.$courseImageLink = $(options.courseImageLink); // Image link to course home page
+                     this.$triggerOpenBtn.on('click', this.toggleSessionSelectionPanel.bind(this));
+
+                     this.render(options);
+                     this.postRender();
```

Comment:
Nit: Would it make sense to call this at the end of `render` so that it always gets called when we re-render?

Reply:
The post render in this case initializes listeners to the document, so we don't want to initialize in every re-render